### PR TITLE
Support drop migration consitency check job

### PR DESCRIPTION
**Pull request by sandynz**
Changes proposed in this pull request:
  - Add DROP MIGRATION CHECK jobId DistSQL and impl
  - Refactor consistency check sequence
  - Add unit test

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Drop migration check statement assert.
+ */
+public final class DropMigrationCheckStatementAssert {
+    
+    /**
+     * Assert stop migration check statement is correct with expected parser result.
```
Comment (RaigorJiang) [inline test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java:34]: `stop` should be `drop`.

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @param assertContext assert context
+     * @param actual actual start migration statement
+     * @param expected expected start migration statement test case
```
Comment (RaigorJiang) [inline test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java:38]: `start` -> `drop` .

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java (hunk 1 above)
Comment (sandynz) [reply test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java]: OK, updated

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java (hunk 2 above)
Comment (sandynz) [reply test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/internal/asserts/statement/ral/impl/migration/update/DropMigrationCheckStatementAssert.java]: OK, updated

**Review by azexcy — APPROVED** [review]

**Pull request merged**